ui/app/pages/confirm-transaction-base/confirm-transaction-base.container.js

```diff
@@ -72,7 +70,7 @@ const mapStateToProps = (state, ownProps) => {
     nonce,
   } = confirmTransaction
   const { txParams = {}, lastGasPrice, id: transactionId, transactionCategory } = txData
-  const transaction = R.find(({ id }) => id === (transactionId || Number(paramsTransactionId)))(selectedAddressTxList) || {}
+  const transaction = Object.values(unapprovedTxs).find(({ id }) => id === (transactionId || Number(paramsTransactionId))) || {}
```

Comment:
Isn't `unapprovedTxs` an array? So this should work:
```suggestion
  const transaction = unapprovedTxs.find(({ id }) => id === (transactionId || Number(paramsTransactionId))) || {}
```

Reply:
my commit of your suggestion might have been hasty... it might be an object

Reply:
yes, appears to be an object

Reply:
confirmed to be an object, comes from `getUnapprovedTxList` in `tx-state-manager` which reduces an array of unapproved txes to an object

Reply:
Whoops! I guess this comment should be updated then: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/transactions/tx-state-manager.js#L77

I looked straight at that, saw the `.reduce`, and it still didn't click that it was an object 🤦‍♂

Reply:
just undid my previous commit of that suggestion with a force push